CPS-0023? | Cardano Multi Asset Treasury (CMAT) (#1103)
cardano-foundation/CIPs: CPS-0023/README.md

---
CPS: 23
Title: Cardano Multi Asset Treasury (CMAT)
Category: Ledger
Status: Open
Authors:
  - Hinson Wong - SIDAN Lab <wongkahinhinson@gmail.com>
  - Felix Weber - Mesh <felix@meshjs.dev>
  - Nicolas Cerny - Cardano Foundation <nicolas.cerny@cardanofoundation.org>
Proposed Solutions: []
Discussions:
  - https://forum.cardano.org/t/cardano-multi-asset-treasury-cmat/149984
  - https://github.com/cardano-foundation/CIPs/pull/1061#issuecomment-3109728207

>   - https://github.com/cardano-foundation/CIPs/pull/1103

Reviewer: ```suggestion
  - Forum discussion: https://forum.cardano.org/t/cardano-multi-asset-treasury-cmat/149984
  - Original PR: https://github.com/cardano-foundation/CIPs/pull/1103
```
I'm deleting the middle comment since it's become less relevant over time as the CMAT discussion has evolved and the Account Address Enhancement has become more widely known.  @lehins if you think this reference should be put back in, then please give us a short label for it:
* https://github.com/cardano-foundation/CIPs/pull/1061#issuecomment-3109728207

Created: 2025-10-26
License: CC-BY-4.0
---

## Abstract

Cardano’s on-chain treasury currently holds funds in ada (lovelace) only. However, there is growing interest in allowing the treasury to support multiple assets. This CPS serves as an initial documents to outline the motivation of Cardano multi asset treasury.

## Problem

The Cardano treasury's exclusive holding of ada (lovelace) creates financial instability for both the treasury and funding proposers. Proposers must denominate their requests in ada, yet their real-world costs are in fiat. This forces them to gamble on a fixed exchange rate in their budgets. Consequently, proposers bear all the financial risk: a drop in ada's price can render their project underfunded and unviable, while a sharp price increase results in an inefficient over-allocation of community funds. This volatility makes sustainable, long-term budgeting impossible for proposers and complicates the treasury's own capital management, creating a high-risk environment that can deter high-quality, long-term ecosystem development.

> ## Use cases

Reviewer: ```suggestion
## Use Cases
```
(if memory serves, validation of header titles requires all words capitalised)

     - Do we need new protocol parameters for CMAT
     - What we use for threshold for listing & delisting if it introduces new governance actions
     - Other ledger rule's technical consideration such as min ADA

   - Update on existing CIPs such as [CIP-1694](https://github.com/cardano-foundation/CIPs/tree/master/CIP-1694).

2. Submit an Info Action of the CIP to gauge community sentiment if this is a requested change.

3. Implementation of the CIP (likely including passing governance action to initial hardfork in updating ledger)

### Out of Scope

- The whitelisted assets to be included in the Cardano Multi Asset Treasury
- The conversion and management of the multi-assets in the treasury

## Open Questions

### Spam Prevention - Only Allow Whitelisted Tokens

Spam prevention as if only whitelisted tokens can send into the treasury. Do we want to build in this feature into CMAT?

- For: Discussion in [(CIP-0159)](https://github.com/cardano-foundation/CIPs/pull/1061) thread acknowledges the need for spam prevention
- Against: Polkadot community mentioned whitelist is not needed as people just do not care

Spam prevention options:

> - Introduce new governance action type with the function to add-new-asset to the treasuy, the gov action would be governed as other gov actions (CIP1694). Also comes with the usual 100k GA deposit.

Author: So with that in mind, would you suggest any change to this draft?

- Another alternative is simply coordinate offchain tool set to ignore non-whitelisted CNTs from indexer, but not enforcing onchain.

### Should Onchain Treasury Participate in DeFi?

> In Polkadot ecosystem, a large part of treasury activities are denominated in stables rather than its native token DOT. In order to meet such demand, Polkadot treausry indeed directly interacting with DeFi project to conduct dollar cost averaging selling DOT into stables. Do we want Cardano treasury to be able to participate in DeFi like Polkadot? Or do we prefer to stay clean and only use CMAT as custody of multi-assets only while conducting most active management offchain?

Reviewer 2: ```suggestion
In Polkadot ecosystem, a large part of treasury activities are denominated in stables rather than its native token DOT. In order to meet such demand, Polkadot treasury indeed directly interacting with DeFi project to conduct dollar cost averaging selling DOT into stables. Do we want Cardano treasury to be able to participate in DeFi like Polkadot? Or do we prefer to stay clean and only use CMAT as custody of multi-assets only while conducting most active management offchain?
```

### Technology Path in Implementation

There are several potential ways to implement CMATs:

1. Cardano Wallet as Treasury

>    > Implementing CMAT through ledger updates which transform the current Cardano Treasury files into an actuall address which can receive, hold and distribute multi assets

Reviewer: Fixed with https://github.com/cardano-foundation/CIPs/pull/1103/commits/770c26375c471fc80b4fce8fd12789382b0636c6 — but the typo corrected above (`actuall`) is still there 😅

     - Any guardrails in practice for inclusion of CNTs
     - Do we define net change limits in CNTs also

   - A new CIP for multi asset treasury covering below areas:

     - Exact specification to describe the ledger change
     - Analysis to take network security into account
     - Do we need new protocol parameters for CMAT
     - What we use for threshold for listing & delisting if it introduces new governance actions
     - Other ledger rule's technical consideration such as min ADA

   - Update on existing CIPs such as [CIP-1694](https://github.com/cardano-foundation/CIPs/tree/master/CIP-1694).

2. Submit an Info Action of the CIP to gauge community sentiment if this is a requested change.

3. Implementation of the CIP (likely including passing governance action to initial hardfork in updating ledger)

### Out of Scope

- The whitelisted assets to be included in the Cardano Multi Asset Treasury
- The conversion and management of the multi-assets in the treasury

## Open Questions

### Spam Prevention - Only Allow Whitelisted Tokens

Spam prevention as if only whitelisted tokens can send into the treasury. Do we want to build in this feature into CMAT?

> - For: Discussion in [(CIP159)](https://github.com/cardano-foundation/CIPs/pull/1061) thread acknowledges the need for spam prevention

Reviewer: ```suggestion
- For: Discussion in [(CIP-0159)](https://github.com/cardano-foundation/CIPs/pull/1061) thread acknowledges the need for spam prevention
```
Please follow the example everywhere else on the repository of either using the tag "CIP-0159" or the name "CIP 159".  Mixing these syntaxes makes text much hard to search.
